app/server.js

```diff
@@ -29,37 +42,92 @@ async function createHttpServer() {
     return
   })
 
-  initialize({
-    app,
-    apiDoc: v1ApiDoc,
-    securityHandlers: {},
-    dependencies: {
-      apiService: v1ApiService,
-    },
-    paths: [path.resolve(__dirname, `api-${API_MAJOR_VERSION}/routes`)],
+  const acapyPathPrefix = `/${API_MAJOR_VERSION}/aca-py`
+  const specTransformOptions = {
+    title: SERVICE_TYPE,
+    version: API_VERSION,
+    pathPrefix: acapyPathPrefix,
+  }
+  app.get(`/${API_MAJOR_VERSION}/api-docs`, async (req, res) => {
+    const apiDocResponse = await swagger2openapi.convertUrl(`${ACAPY_ADMIN_SERVICE}/api/docs/swagger.json`, {
+      patch: true,
+    })
+    const apiDoc = apiDocResponse.openapi
+    // TODO: cache this
+    res.send(await transformApiDoc(apiDoc, specTransformOptions))
   })
 
   const options = {
     swaggerOptions: {
       urls: [
         {
-          url: `http://localhost:${PORT}/${API_MAJOR_VERSION}/api-docs`,
-          name: 'ApiService',
+          url: `http://${SERVICE_HOST}:${SERVICE_PORT}/${API_MAJOR_VERSION}/api-docs`,
+          name: SERVICE_TYPE,
         },
       ],
     },
   }
 
   app.use(`/${API_MAJOR_VERSION}/swagger`, swaggerUi.serve, swaggerUi.setup(null, options))
 
-  // Sorry - app.use checks arity
+  // auth0 middleware for authenticated routes
+  const checkJwt = auth({
+    audience: AUTH_AUDIENCE,
+    issuerBaseURL: AUTH_ISSUER,
+  })
+  app.use(acapyPathPrefix, checkJwt)
+
+  // wallet create
+  app.post(`${acapyPathPrefix}/multitenancy/wallet`, async (req, res) => {
+    const subject = req.auth.payload.sub
+    const wallets = await querySubWallets(subject)
+    if (wallets.length !== 0) {
+      res.status(409).type('text/plain').send(`409 Wallet already exists`)
+    }
+
+    const response = await createSubWallet({
+      ...req.body,
+      wallet_name: subject,
+    })
+
+    const result = Buffer.from(await response.arrayBuffer())
+    res.status(response.status).type(response.headers.get('content-type')).send(result)
+  })
+
+  // other paths
+  app.all(`${acapyPathPrefix}/*`, async (req, res) => {
```

Comment:
mmm... maybe a middleware and `app.use(authMiddleware)` again this is just a suggestion I don't anything against any method but middleware would make server.js easier to read at least for me also I might be missing some elements here, but where `auth0` token comes in?

Reply:
agreed 😄 , see ln 78:

```js
const checkJwt = auth({
  audience: AUTH_AUDIENCE,
  issuerBaseURL: AUTH_ISSUER,
})
app.use(acapyPathPrefix, checkJwt)
```

Reply:
I have noticed that I didn't have any tests to prove it does `401` without authentication. Added